### ISSUE TITLE
Upgrade `prefer_void_to_null` to stable

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#1287]
@MichaelRFairhurst: what needs to happen here?

Looking at uses,

https://github.com/search?q=prefer_void_to_null+filename%3Aanalysis_options.yaml&type=Code

a number of files include it while still other have it removed:

```
# TODO: Re-enable prefer_void_to_null when fully supported
#- prefer_void_to_null
```

Thoughts on next steps?

**Comment by a14n** [comment]
FWIW [Flutter already uses this lint](https://github.com/flutter/flutter/blob/5391447fae6209bb21a89e6a5a6583cac1af9b4b/analysis_options.yaml#L142).

**Comment by pq** [comment]
@MichaelRFairhurst: thoughts on this?  I'm inclined to bump to `stable` unless you're actively working on anything that you think should keep this flagged `experimental`?

**Comment by MichaelRFairhurst** [comment]
Nope, upping to stable sgtm!
